### PR TITLE
[pydocs] update sortMethods to only suggest using string

### PR DESCRIPTION
**Pull request by MartijnKaijser**

### CODE REVIEW

**Comment by Montellese** [comment]
How is this supposed to work? The strings would have to be defined somewhere (probably again as constants) and we would have to change the implementation to handle strings instead of ints. But I don't see why the ints are a problem. Is anybody using the actual integer values? If so they should move to the existing constants but I wouldn't touch the constants definitions.

**Comment by MartijnKaijser** [comment]
it already works.
http://xbmclogs.com/ptjsjxwyn/vlcvps/raw

**Comment by phil65** [comment]
It doesnt work. The stuff you posted passes ints for sortmethod, not strings. ;)

**Comment by ronie** [comment]
change looks ok to me now.

**Comment by phil65** [comment]
no, not ok. the documentation is wrong now.

**Comment by ronie** [comment]
@phil65 could please make a comment on the current diff on the exact line that is incorrect?

File: xbmc/interfaces/legacy/ModuleXbmcplugin.h [hunk 1]
```diff
@@ -91,21 +91,60 @@ namespace XBMCAddon
...
-     * sortMethod  : integer - number for sortmethod see SortFileItem.h.\n
-     * label2Mask  : [opt] string - the label mask to use for the second label.  Defaults to '%D'
-     *               - applies to:
-     *                           - SORT_METHOD_NONE, SORT_METHOD_UNSORTED, SORT_METHOD_VIDEO_TITLE,
-     *                           - SORT_METHOD_TRACKNUM, SORT_METHOD_FILE, SORT_METHOD_TITLE,
-     *                           - SORT_METHOD_TITLE_IGNORE_THE, SORT_METHOD_LABEL,
-     *                           - SORT_METHOD_LABEL_IGNORE_THE, SORT_METHOD_VIDEO_SORT_TITLE,
-     *                           - SORT_METHOD_VIDEO_SORT_TITLE_IGNORE_THE, SORT_METHOD_FULLPATH,
-     *                           - SORT_METHOD_LABEL_IGNORE_FOLDERS, SORT_METHOD_CHANNEL
+     * sortMethod  : string - sortmethod see SortFileItem.h.\n
```
Comment (phil65) [inline xbmc/interfaces/legacy/ModuleXbmcplugin.h]: the type of sortMethod doesnt change. It is an int, changing the docs doesnt make it a string. ;)

Comment (ronie) [inline xbmc/interfaces/legacy/ModuleXbmcplugin.h]: in your python code, you pass 'xbmcplugin.SORTMETHOD_DATEADDED' to the addSortMethod function.
'xbmcplugin.SORTMETHOD_DATEADDED' is not an integer, it's a string.
the fact that kodi internally matches this to an int is not relevant for our python documentation imo.

Comment (phil65) [inline xbmc/interfaces/legacy/ModuleXbmcplugin.h]: you are wrong. at no point of time xbmcplugin.SORTMETHOD_DATEADDED is a string. It´s always an int.

Comment (phil65) [inline xbmc/interfaces/legacy/ModuleXbmcplugin.h]: You can check by doing "type(xbmcplugin.SORTMETHOD_DATEADDED)" in python.

Comment (MartijnKaijser) [inline xbmc/interfaces/legacy/ModuleXbmcplugin.h]: i don't care what you call it. it's an alien for all i care

Comment (ronie) [inline xbmc/interfaces/legacy/ModuleXbmcplugin.h]: ah, gotcha, thx :-)

Comment (phil65) [inline xbmc/interfaces/legacy/ModuleXbmcplugin.h]: sorry, but you´re probably misunderstanding something. No need to get offensive.
Another try: The xbmcplugin class SWIG generates basically looks like this:
```python
 class xbmcplugin(object):
    self.SORTMETHOD_DATEADDED = 1
    self.SORTMETHOD_xxx = 2
    ...

    def __init(self, *args, **kwargs):
    ...
```

**Comment by phil65** [comment]
lookin fine now.

**Comment by Montellese** [comment]
Yup now it's fine.

**Comment by MartijnKaijser** [comment]
next time explain yourself more clear

**Comment by phil65** [comment]
friendly as usual. ;)
they were clear, from first comment to last comment.

**Comment by MartijnKaijser** [comment]
apparently not as ronie didn't get your point either

**Comment by phil65** [comment]
ronie probably is able to recognize though that he had some kind of "brainfart" moment, which we all have sometimes.  His last response was "ah, gotcha, thx :-)" btw.

**Pull request merged**